### PR TITLE
Link Classroom to Classroom Desktop

### PR DESCRIPTION
**Pull request by srinjoym**
Continues #39 (switched branch from fork)

I have split up this PR into three parts so that it is easier to review. Please put all comments on the smaller PRs.

1. Module Updates on Renderer  https://github.com/education/classroom-desktop/pull/41
2. UI Updates for renderer https://github.com/education/classroom-desktop/pull/42
3. Main Process updates + OAuth Changes https://github.com/education/classroom-desktop/pull/43

This PR is the combination of all three and has all commit history.

### CODE REVIEW

File: app/lib/cloneutils.js [hunk 1]
```diff
@@ -46,7 +46,10 @@ export const clone = (repoURL, destination, progressCallback, token) => {
     }
 
     if (token) {
-      options.fetchOpts.callbacks.credentials = function () {
+      options.fetchOpts.callbacks.certificate_check = () => { // for libgit2 problem
```
Comment (srinjoym) [inline app/lib/cloneutils.js]: Configuration for nodegit to clone with access token, will shift to using electron bindings in the future for better performance

File: app/main.js [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
 let mainWindow
+let deepLinkURLOnReady = null
 
 logger.init()
 
 function createWindow () {
   logger.info("creating app window")
 
+  app.setAsDefaultProtocolClient("x-github-classroom")
+  const menu = defaultMenu(app, shell)
```
Comment (srinjoym) [inline app/main.js]: Sets mac menu options so users can copy, paste and get default menu options in app.

**Comment by srinjoym** [comment]
@johndbritton @shiftkey Hi guys 👋 I've split this PR up into three smaller PRs just so that it's easier to review, would love some feedback on the code!

**Comment by shiftkey** [comment]
>  Please put all comments on the smaller PRs.

Sigh, I glossed over this comment. Let's see what I can do to port those over now.

**Comment by srinjoym** [comment]
No need to port the comments! I just made the smaller PRs so that it'd be easier to look at the code, feel free to make comments wherever it's easier for you

**Comment by shiftkey** [comment]
@srinjoym it's all good - I see you've been adding comments to the smaller PRs so I'll do the same 🤘

**Comment by shiftkey** [comment]
@srinjoym another benefit of keeping discussions in the smaller PRs is that, if you're pushing changes to those PRs to address feedback, the feedback will collapse and it's easier to track the remaining feedback, rather than also having this monolithic PR.

**Comment by d12** [comment]
@srinjoym Unfortunately I have no electron experience, I won't be able to review this code in any meaningful way.

When this is ready to be built though, I'm happy to test things out in a few different environments.

**Comment by srinjoym** [comment]
@shiftkey I've addressed your first pass comments, let me know if everything looks good!

**Comment by shiftkey** [comment]
@srinjoym if we can merge the smaller PRs independent of each other (or at least in order #41, #42, then #43) can we use those and close this out to avoid further confusion?

**Comment by srinjoym** [comment]
@shiftkey Yea, we can definitely merge those independently now, I'll close this issue and paste the description from this into the smaller PRs.

**Pull request closed without merging**